packages/trip-details/src/fare-detail.tsx

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable import/prefer-default-export */
+import { Leg, Money } from "@opentripplanner/types";
+
+import React from "react";
+import styled from "styled-components";
+import { FormattedNumber } from "react-intl";
+import { FareDetailsProps } from "./types";
+
+interface FareTypeTableProps {
+  header: string;
+  cols: {
+    key: string;
+    header: string;
+  }[];
+  legs: (Leg & { fares: { [key: string]: { price: Money } } })[];
+  fareTotals: { [fareKey: string]: Money };
+}
+
+const TableHeader = styled.thead`
+  th {
+    min-width: 5ch;
+    padding: 0.75em 1.5em;
+    text-align: center;
+  }
+  th:nth-of-type(2n + 1) {
+    background: #cccccc22;
+  }
+  th.main {
+    color: #ffffffcc;
+    background: #333333;
+  }
+`;
+
+const TotalRow = styled.tr`
+  border-top: 2px solid #333;
+  font-weight: 600;
+`;
+
+const Table = styled.table`
+  border-collapse: collapse;
+  display: block;
+  font-family: sans-serif;
+  margin: -8px;
+  /* This line can not be sorted alphabetically, because it must come after the margin rule */
+  margin-bottom: 16px;
+  padding: 0;
+
+  td {
+    text-align: right;
+  }
+  td:nth-of-type(2n + 1) {
+    background: #cccccc22;
+  }
+  td.no-zebra {
+    background: transparent;
+  }
+  th:first-of-type {
+    height: 40px;
+  }
+`;
+
+const FareTypeTable = (props: FareTypeTableProps): JSX.Element => {
+  const { header, fareTotals, cols, legs } = props;
+  return (
+    <Table>
+      <TableHeader>
+        <th className="main">{header}</th>
+        {cols.map(col => (
+          <th key={col.header}>{col.header}</th>
+        ))}
+      </TableHeader>
+      {legs.map((leg, index) => (
+        <tr key={index}>
+          <td className="no-zebra">{leg.routeShortName}</td>
+          {cols.map(col => {
+            const fare = leg.fares[col.key];
+            return (
+              <td key={col.key}>
+                <FormattedNumber
+                  value={fare?.price?.cents / 100 || 0}
```

Comment:
What do you think about displaying "free" or something that indicates it is part of a free transfer?

Reply:
I think this is a good idea what do you think @daniel-heppner-ibigroup

Reply:
Agree, I like this idea. It will need to be internationalized however.

Reply:
We aren't actually using the `isTransfer` information that we get from OTP, so that would make a lot of sense to use. That would let us differentiate between a free service and a free transfer. @miles-grant-ibigroup

Reply:
How about using an icon like `@styled-icons/boxicons-regular/Transfer` next to fares that use a transfer?

Reply:
I love that! Great idea